### ISSUE TITLE
[CRASH] Async HTTPS crashes

### ISSUE THREAD

**Issue by bcnewlin** [OpenSIPS/opensips#1776]
**OpenSIPS version you are running**

```
# opensips -V
version: opensips 2.4.6 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, F_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 4e258b17f
main.c compiled on 19:02:09 Jul 22 2019 with gcc 7
```

**Crash Core Dump**

https://pastebin.com/kr7B6QHT

**Describe the traffic that generated the bug**

Based on logs, it appears to be due to the use HTTPS with the rest_client module.

**To Reproduce**

1. Start OpenSIPS
2. Initiate a rest_client HTTPS call that times out
3. Check OpenSIPS crashed

**Relevant System Logs**

```
Jul 23 13:51:23 [6994] ERROR:rest_client:resume_async_http_req: connected, but transfer timed out (5s)
Jul 23 13:51:24 [6994] CRITICAL:core:sig_usr: segfault in process pid: 6994, id: 21
Jul 23 13:53:27 [6942] INFO:core:handle_sigs: child process 6994 exited by a signal 11
Jul 23 13:53:32 [6942] CRITICAL:core:fm_status: different free frag. count: 34!=29 for hash  75


**OS/environment information**
 - Operating System: <!-- (example: `Debian 9`) --> Amazon Linux 2
 - OpenSIPS installation: <!-- (example: `git`/`source`/`debs`/`manual packages`) --> git
 - other relevant information:

**Additional context**
<!-- Add any other context about the problem here. -->

**Comment by liviuchircu** [comment]
There was a recent memory-management fix that went into rest_client: 76625354cfd3.  Can you try and see if you can reproduce this issue with latest `2.4`, possibly using the nightly packages?  Thanks!

**Comment by bcnewlin** [comment]
Thanks for the quick response! We have attempted to pick up the latest 2.4 but have been blocked by crashes reported in #1771 . I will try to pull your commit specifically and see if it addresses the issue.
